motoko/internet_identity_integration/src/greet_frontend/src/index.js

```diff
@@ -1,48 +1,61 @@
+AuthClient.create().then((client) => {
+  // Put this on the window so we can access it from the console. This is helpful for learning, but it's recommended to keep it in a closure in a real app.
+  window.authClient = client;
+  // We can use the authClient to check if the user is already authenticated
+  setActor(client);
+
+  // Now that the auth client is created, we can enable the login and greet buttons
+  loginButton.removeAttribute("disabled");
+  greetButton.removeAttribute("disabled");
+});
+
+loginButton.onclick = () => {
+  // start the login process and wait for it to finish
+  window.authClient.login({
+    identityProvider: process.env.II_URL,
+    onSuccess: () => {
+      setActor(window.authClient);
+    },
+    onError: (err) => {
+      console.error(err);
+    },
+  });
+};
 
-    // Interact with backend actor, calling the greet method
-    const greeting = await actor.greet();
+greetButton.onclick = async () => {
+  // Disable the button to prevent multiple clicks
+  greetButton.setAttribute("disabled", true);
 
-    greetButton.removeAttribute("disabled");
+  // Interact with backend actor, calling the greet method
+  const greeting = await window.greetActor.greet();
 
-    document.getElementById("greeting").innerText = greeting;
+  greetButton.removeAttribute("disabled");
 
-    return false;
+  document.getElementById("greeting").innerText = greeting;
 };
 
-const loginButton = document.getElementById("login");
-loginButton.onclick = async (e) => {
-    e.preventDefault();
-
-    // create an auth client
-    let authClient = await AuthClient.create();
-
-    // start the login process and wait for it to finish
-    await new Promise((resolve) => {
-        authClient.login({
-            identityProvider: process.env.II_URL,
-            onSuccess: resolve,
-        });
-    });
-
-    // At this point we're authenticated, and we can get the identity from the auth client:
-    const identity = authClient.getIdentity();
-    // Using the identity obtained from the auth client, we can create an agent to interact with the IC.
-    const agent = new HttpAgent({identity});
-    // Using the interface description of our webapp, we create an actor that we use to call the service methods.
-    actor = createActor(process.env.GREET_BACKEND_CANISTER_ID, {
-        agent,
-    });
-
-    return false;
-};
\ No newline at end of file
+/**
+ * Sets the actor to be used by the greet button
+ * @param {AuthClient} an initialized AuthClient, which will have the identity of the user (logged in or not)
+ */
+function setActor(authClient) {
+  // At this point we're authenticated, and we can get the identity from the auth client:
+  const identity = authClient.getIdentity();
+  // Using the identity obtained from the auth client, we can create an agent to interact with the IC.
+  const agent = new HttpAgent({ identity });
+  // Using the interface description of our webapp, we create an actor that we use to call the service methods.
```

Comment:
Should we also mention that this shouldn't be exposed in `window`?

Reply:
I added that info to the comment where the global variables were declared